library/edi.inc

```diff
@@ -140,7 +142,7 @@ function create_NM1($row, $nm1Cast, $X12info, $segTer, $compEleSep)
         $NM1[5] = "";                       // Data Element not required.
         $NM1[6] = "";                       // Data Element not required.
         $NM1[7] = "";                       // Data Element not required.
-        $NM1[8] = $X12info['id_number'] ? $X12info['id_number'] : "46";  // 46 - Electronic Transmitter Identification Number (ETIN)
+        $NM1[8] = "PI";                     // 5010 no longer uses "46"
```

Comment:
While this may work in most or some cases, depending on whom is target, any item that is configurable from x12 partner should override hard coded values. i.e Should avoid hard coding options.
Default from x12 partner for this item is empty so setting default in conditional to FI will achieve what you need but still allow changing in partner setup.
In reality, more x12 segment config options should be added to x12 partner setup but for now, can we please still allow what configuration options are available during partner setup.

Reply:
sure, makes sense but PI is a better default
id number isn't a good fit here though since it usually is an important number like a tax id that can be used to fill a value that isn't filled in the current config

Reply:
am getting to the ui part but first have to see if can tie in your parsing of the 271 for batch eligibility requests to be presentable in the demographics like your real time

Reply:
Actually I meant PI, sorry. In the case of NM1:8 it represents the code qualifier for NM1:9 and can be different depending on partner. Also, we can't be sure everyone is currently complying with 46 no longer supported so user may need to override new default. I fear Stephen this is just me being overly cautious and I have the burn scars to prove it:)

Reply:
For ui I tested against batches and should result in the same report. What's the issue?

Reply:
after you upload the 271 nothing shows in demographics

Reply:
this is on 5.0.1(7)

Reply:
just says the following file has been uploaded

Reply:
hmm, so you must be using the patch I published on the forum. Let me test this again and see what's to see. I'll get back to ya...

Reply:
For this I did some research of about 25 insurance companies (I have an edi exchange companion guide that covers at least 50 payers) and they all list PI for the 2100A loop so it'd be okay to leave this the way you have it.
We might need to use that partner setup field to address the darn 2100C loop DTP03 service date